docs/patterns/tools/use-tools/processing-actions.mdx

```diff
@@ -127,12 +96,112 @@ agent_executor = AgentExecutor(agent=agent, tools=tools, verbose=True)
+```typescript TypeScript
+const schemaProcessor: TSchemaProcessor = ({
+  actionName,
+  toolSchema,
+}: {
+  actionName: string;
+  toolSchema: RawActionData;
+}) => {
+  const modifiedSchema = { ...toolSchema };
+  modifiedSchema.parameters = {
+    ...modifiedSchema.parameters,
+    required: modifiedSchema.parameters?.required?.filter(
+      field => !['project_id', 'team_id'].includes(field)
+    ) || []
+  };
+
+  return modifiedSchema;
+};
+```
+</Step><Step title="Input Modifier">
+This will be used to modify the input data for the `LINEAR_CREATE_LINEAR_ISSUE` action. Here we have added the values for `project_id` and `team_id` parameters to the input data. By doing this, we can avoid specifying these values in the prompt and be sure that the agent uses the correct values. The technical term for this is **Pre-Processing**.
+```typescript TypeScript
+const preProcessor: TPreProcessor = ({ params, actionName, appName }: {
+  params: Record<string, unknown>;
+  actionName: string;
+  appName: string;
+}) => {
+  const modifiedParams = { ...params };
+
+  modifiedParams.project_id = "e708162b-9b1a-4901-ab93-0f0149f9d805";
+  modifiedParams.team_id = "249ee4cc-7bbb-4ff1-adbe-d3ef2f3df94e";
+
+  return modifiedParams;
+}
+```
+</Step>
+<Step title="Output Modifier">
+This will be used to modify the output data for the `LINEAR_CREATE_LINEAR_ISSUE` action. Here we are modifying the output to just return the action execution status `successful` & the `issueId`, by doing this can keep the LLM context clean. The technical term for this is **Post-Processing**.
+```typescript TypeScript
+const postProcessor: TPostProcessor = ({ actionName, appName, toolResponse }: {
+  actionName: string;
+  appName: string;
+  toolResponse: ActionExecutionResDto;
+}) => {
+  const issueId = toolResponse.data.id;
+  return { data: { id: issueId }, successful: true };
+}
+```
+</Step>
+<Step title="Add the Processors to Toolset & Execute the Agent">
+After creating the processors, we need to add them to the toolset using `addSchemaProcessor`, `addPreProcessor`, and `addPostProcessor` methods and then get the tools. Last we create the agent and execute it.
+```typescript TypeScript {2-4}
+async function main() {
+  toolset.addSchemaProcessor(schemaProcessor);
+  toolset.addPreProcessor(preProcessor);
+  toolset.addPostProcessor(postProcessor);
+
+  const tools = await toolset.getTools({
+    actions: ["LINEAR_CREATE_LINEAR_ISSUE"]
+  });
+
+  const prompt = (await pull(
+    "hwchase17/openai-functions-agent"
+  )) as ChatPromptTemplate;
+
+  const agent = await createOpenAIFunctionsAgent({
+    llm,
+    tools,
+    prompt,
+  });
+
+  const agentExecutor = new AgentExecutor({ agent, tools, verbose: true });
+
+  const response = await agentExecutor.invoke({ input: "Create an issue on linear to update the frontend to with new design and description 'Refer to the design in the attachment', estimate is 5 (L) & return the issue id" });
```

Comment:
There's a minor grammatical error in the example task string: "to update the frontend to with new design" should be "to update the frontend with new design"